src/plots/cartesian/autorange.js

```diff
@@ -95,76 +102,83 @@ function getAutoRange(gd, ax) {
         for(j = 0; j < maxArray.length; j++) {
             maxpt = maxArray[j];
             dv = maxpt.val - minpt.val;
-            dp = ax._length - getPad(minpt) - getPad(maxpt);
-            if(dv > 0 && dp > 0 && dv / dp > mbest) {
-                minbest = minpt;
-                maxbest = maxpt;
-                mbest = dv / dp;
+            if(dv > 0) {
+                dp = axLen - getPad(minpt) - getPad(maxpt);
+                if(dp > minSpan) {
+                    if(dv / dp > mbest) {
+                        minbest = minpt;
+                        maxbest = maxpt;
+                        mbest = dv / dp;
+                    }
+                }
+                else if(dv / axLen > mbest) {
+                    // in case of padding longer than the axis
+                    // at least include the unpadded data values.
+                    minbest = {val: minpt.val, pad: 0};
+                    maxbest = {val: maxpt.val, pad: 0};
+                    mbest = dv / axLen;
```

Comment:
@etpinard we had an issue recently where someone used ridiculously large markers and autorange broke, didn't we? I can't find it, but this fixes that case, by trying again with no padding.

Reply:
Ah, yep - I thought it was more recent but that seems like the one. Are you happy with this solution, enough that we should consider #1397 closed with this PR?

Reply:
```
var trace1 = {
  x: [10000, 0, 30, 40000],
  y: [6000, 0, 80, 600],
  mode: 'markers',
  marker: {
    size: [70000, 0, 30, 9000],
    //sizeref: 100
  }
};

var data = [trace1];

var layout = {
  title: 'Marker Size',
  showlegend: false,
  height: 600,
  width: 600
};

Plotly.newPlot('graph', data, layout);
```

now renders as 

![image](https://user-images.githubusercontent.com/6675409/45175135-a429f480-b1da-11e8-9c86-5f0d0728e928.png)

which I would call _correct_.

With a "reasonable" `marker.sizeref` of 500, we get

![image](https://user-images.githubusercontent.com/6675409/45175196-ce7bb200-b1da-11e8-8d62-93aeb5f669b6.png)

where `getAutoRange` did its job.